### ISSUE TITLE
Calling on_entry only after saving the entry in the json file

### ISSUE THREAD

**Issue by lorenzoPrimi** [measurement-kit/measurement-kit#1109]
Needed for the mobile app

**Comment by bassosimone** [comment]
This is not simple to implement, because the report layer reports an error if it cannot write the entry on the file system *or* it cannot update the remote report. So, it is not as simple as moving the code around, as I initially believed we could do when we discussed this issue. As such, it cannot land in stable. Therefore, I will reschedule this issue for the next release cycle.

**Comment by bassosimone** [comment]
This is problematic: in the current model, `on_entry` is called *in any case* and I would like it to stay like this, because it's an opportunity for who uses the app to deal with the entry the way he/she likes the most and because it's orthogonal from saving on disk. With the new proposed behavior, the entry would be called only when saving on disk and/or submitting remotely succeeded. I see this as pejorative and not as ameliorative.

Currently, there is the `ignore_write_entry_error` that defaults to `0`. Setting it to `1` would interrupt the test if we cannot either write on disk or remotely submit the entry. I guess setting this variable to `1` would be excessive, because we probably want the test to continue even though the report cannot be submitted.

Am I understanding it correctly that your original concern is that you would like to know when an entry was not correctly written on disk? I'd say in this case we'd better abort the test, because we should assume that, if we want to write on disk and we cannot, either it's fully or we don't have permissions, no?

**Comment by bassosimone** [comment]
The original issue is a `wontfix`. So, I applied the corresponding label. However, it seems to me there is room for improvement in MK for addressing the underlying concern (i.e. "I am integrating MK and I want to know when writing to disk fails"). I'll wait for @lorenzoPrimi to reply and, if my understanding of the underlying concern is correct, I guess we can rename the issue and find another solution.

**Comment by bassosimone** [comment]
I discussed this on slack with @lorenzoPrimi and we agreed to close it as wontfix.
